src/modules/admin/components/forms/EditFormButton.tsx

```diff
@@ -0,0 +1,82 @@
+import DashboardCustomizeIcon from '@mui/icons-material/DashboardCustomize';
+import { LoadingButton } from '@mui/lab';
+import { useCallback } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { cache } from '@/providers/apolloClient';
+import { AdminDashboardRoutes } from '@/routes/routes';
+import {
+  FormIdentifierDetailsFragment,
+  useCreateDraftFormMutation,
+} from '@/types/gqlTypes';
+import { generateSafePath } from '@/utils/pathEncoding';
+
+export enum FormEditorType {
+  FormBuilder,
+  JsonEditor,
+}
+
+interface EditFormButtonProps {
+  formIdentifier: FormIdentifierDetailsFragment;
+  text?: string;
+  Icon?: React.ComponentType;
+  editorRoute?: FormEditorType;
+}
+
+const EditFormButton: React.FC<EditFormButtonProps> = ({
+  formIdentifier,
+  text,
+  Icon,
+  editorRoute = FormEditorType.FormBuilder,
+}) => {
+  const navigate = useNavigate();
+
+  const goToEditor = useCallback(
+    (draftId: string) => {
+      const route =
+        editorRoute === FormEditorType.JsonEditor
+          ? AdminDashboardRoutes.JSON_EDIT_FORM
+          : AdminDashboardRoutes.EDIT_FORM;
+
+      navigate(
+        generateSafePath(route, {
+          identifier: formIdentifier.identifier,
+          formId: draftId,
+        })
+      );
+    },
+    [formIdentifier, editorRoute, navigate]
+  );
+
+  const [createDraftForm, { loading, error }] = useCreateDraftFormMutation({
```

Comment:
I think we may need to move this up into the parent component so that each button can be blocked while the other is loading. We don't want the user to be able to click "Edit Form" and then "Json Editor" in rapid succession, leading to an error message since two drafts can't both exist at the same time.

However, moving it up into the parent proved a little difficult because of the `goToEditor` callback, so I put it aside for now; will return with fresh eyes tomorrow

Reply:
The new [change from the backend PR](https://github.com/greenriver/hmis-warehouse/pull/4449#discussion_r1636773496) where we just return the existing draft instead of throwing an error kinda solves this problem, I think. Also, since the JSON editor is only accessible to staff level admins, I'd propose to not put any more time into it now.